packages/kit/src/runtime/client/types.d.ts

```diff
@@ -1,12 +1,10 @@
 import { CSRComponent, CSRRoute, NormalizedLoadOutput } from 'types/internal';
-import { Page } from 'types/page';
 
 export type NavigationInfo = {
 	id: string;
 	routes: CSRRoute[];
-	path: string;
-	decoded_path: string;
-	query: URLSearchParams;
+	url: URL;
+	// decoded_path: string; TODO do we need this?
```

Comment:
Without `decoded_path` to cache the result of the decoding we're having to do the decoding twice now. Not sure if that's a concern or not. I'd say we should either restore this for the performance enhancement or remove this line and the TODO

Reply:
yeah i was working on this as you were typing the comment :)